### ISSUE TITLE
[Bug]: FillLayer not rendered on IOS

### ISSUE THREAD

**Issue by florian-raab** [rnmapbox/maps#3400]
### Mapbox Implementation

Mapbox

### Mapbox Version

default

### React Native Version

0.73.2

### Platform

iOS

### `@rnmapbox/maps` version

10.1.16

### Standalone component to reproduce

```javascript
import React from 'react';
import {
  MapView,
  ShapeSource,
  FillLayer, 
  Camera,
} from '@rnmapbox/maps';
import circle from '@turf/circle'

class BugReportExample extends React.Component {
  render() {
    return (
      <MapView style={{flex: 1}}>
        <Camera centerCoordinate={[-74.00597, 40.71427]} zoomLevel={14} />
        <ShapeSource id={'CircleShape'}
                              shape={
                                  circle([-74.00597, 40.71427],
                                    200,
                                    {steps: 100, units: 'meters', properties: null})
                              }>
            <FillLayer id={'FillLayer'}
                             style={{
                                 fillColor: '#FF0000',
                             }}/>
        </ShapeSource>
      </MapView>
    );
  }
}
```

### Observed behavior and steps to reproduce

The shape with the fill layer doesn't get rendered on the map and an error is thrown (Only on IOS).

![image](https://github.com/rnmapbox/maps/assets/102097548/19e9280b-9339-441f-addc-dcc1e68bf567)

### Expected behavior

The map should render a red circle with a radius of 200m at [-74.00597, 40.71427].
![image](https://github.com/rnmapbox/maps/assets/102097548/f8481b9c-0910-4d95-a8d0-099c0455d048)

### Notes / preliminary analysis

Inspecting the React-Native Logs:
`ERROR  Mapbox [error] inserting layer failed at position default: Error parsing layer FillLayer of type: fill`
Other Layers (like LineLayer) work fine in the ShapeSource.

This bug is not present in 10.1.15 and before.

**Comment by erwanlpfr** [comment]
Hello, 

I have exactly the same issue, same logs and same behavior.
Present also in `10.1.17`.

**Comment by mfazekas** [comment]
Seems to be a regression in recent rnmapbox. Fixed in: #3402


<img width="398" alt="image" src="https://github.com/rnmapbox/maps/assets/52435/2b6ba14e-ffa7-4287-9edf-a4afe2683bb3">
